gnovm/tests/file.go

```diff
@@ -586,23 +595,37 @@ func (tb *testBanker) SendCoins(from, to crypto.Bech32Address, amt std.Coins) {
 	tb.coinTable[from] = frest
 	// Second, add to 'to'.
 	// NOTE: even works when from==to, due to 2-step isolation.
-	tcoins, _ := tb.coinTable[to]
+	tcoins := tb.coinTable[to]
 	tsum := tcoins.Add(amt)
 	tb.coinTable[to] = tsum
 }
 
 func (tb *testBanker) TotalCoin(denom string) int64 {
-	panic("not yet implemented")
+	return tb.totalCoin[denom]
 }
 
 func (tb *testBanker) IssueCoin(addr crypto.Bech32Address, denom string, amt int64) {
-	coins, _ := tb.coinTable[addr]
+	coins := tb.coinTable[addr]
 	sum := coins.Add(std.Coins{{denom, amt}})
 	tb.coinTable[addr] = sum
+
+	totalCoin, ok := overflow.Add64(tb.totalCoin[denom], amt)
+	if !ok {
+		panic(fmt.Sprintf("totalCoin overflow/underflow for denom %s while adding %d", denom, amt))
+	}
+
+	tb.totalCoin[denom] = totalCoin
 }
 
 func (tb *testBanker) RemoveCoin(addr crypto.Bech32Address, denom string, amt int64) {
-	coins, _ := tb.coinTable[addr]
+	coins := tb.coinTable[addr]
 	rest := coins.Sub(std.Coins{{denom, amt}})
 	tb.coinTable[addr] = rest
+
+	totalCoin, ok := overflow.Sub64(tb.totalCoin[denom], amt)
+	if !ok {
+		panic(fmt.Sprintf("totalCoin overflow/underflow for denom %s while removing %d", denom, amt))
```

Comment:
Nitpick: here we can already distinguish whether it's overflow or underflow, right?

Reply:
https://github.com/gnolang/gno/pull/2671/files#diff-1b9622e2b0f29c9d78624c17933bc65abb5012162a1868ae0ac596dab8891db6L598-R625

Reply:
the prev functions i used actually didn't bring panic info